go.mod

```diff
@@ -1,124 +1,119 @@
 
-go 1.12
+go 1.13
 
 require (
-	cloud.google.com/go v0.37.2 // indirect
-	contrib.go.opencensus.io/exporter/ocagent v0.4.12 // indirect
-	github.com/Azure/go-autorest v11.7.0+incompatible // indirect
-	github.com/DATA-DOG/go-sqlmock v1.3.3 // indirect
-	github.com/MakeNowJust/heredoc v0.0.0-20171113091838-e9091a26100e // indirect
-	github.com/Masterminds/goutils v1.1.0 // indirect
-	github.com/Masterminds/semver v1.4.2 // indirect
-	github.com/Masterminds/sprig v0.0.0-20190301161902-9f8fceff796f // indirect
-	github.com/PuerkitoBio/purell v1.1.1 // indirect
 	github.com/blang/semver v3.5.1+incompatible
-	github.com/chai2010/gettext-go v0.0.0-20170215093142-bf70f2a70fb1 // indirect
+	github.com/containerd/containerd v1.3.0 // indirect
 	github.com/coreos/bbolt v1.3.3 // indirect
-	github.com/coreos/go-semver v0.2.0
-	github.com/coreos/prometheus-operator v0.29.0
-	github.com/cyphar/filepath-securejoin v0.2.2 // indirect
-	github.com/docker/spdystream v0.0.0-20181023171402-6480d4af844c // indirect
+	github.com/coreos/go-semver v0.3.0
+	github.com/coreos/prometheus-operator v0.33.0
 	github.com/elazarl/goproxy v0.0.0-20190421051319-9d40249d3c2f // indirect
 	github.com/elazarl/goproxy/ext v0.0.0-20190421051319-9d40249d3c2f // indirect
-	github.com/exponent-io/jsonpath v0.0.0-20151013193312-d6023ce2651d // indirect
-	github.com/fatih/camelcase v1.0.0 // indirect
-	github.com/ghodss/yaml v1.0.0
+	github.com/ghodss/yaml v1.0.1-0.20180820084758-c7ce16629ff4
 	github.com/go-logr/logr v0.1.0
 	github.com/go-logr/zapr v0.1.1
-	github.com/go-openapi/swag v0.19.0 // indirect
-	github.com/go-sql-driver/mysql v1.4.1 // indirect
-	github.com/gobuffalo/packr v1.30.1 // indirect
-	github.com/gobwas/glob v0.2.3 // indirect
-	github.com/google/uuid v1.1.1 // indirect
-	github.com/grpc-ecosystem/go-grpc-middleware v1.0.0 // indirect
-	github.com/hashicorp/golang-lru v0.5.1 // indirect
-	github.com/huandu/xstrings v1.2.0 // indirect
+	github.com/gosuri/uitable v0.0.3 // indirect
 	github.com/iancoleman/strcase v0.0.0-20180726023541-3605ed457bf7
-	github.com/jmoiron/sqlx v1.2.0 // indirect
-	github.com/lib/pq v1.1.1 // indirect
-	github.com/mailru/easyjson v0.0.0-20190403194419-1ea4449da983 // indirect
 	github.com/markbates/inflect v1.0.4
 	github.com/martinlindhe/base36 v0.0.0-20180729042928-5cda0030da17
 	github.com/mattbaird/jsonpatch v0.0.0-20171005235357-81af80346b1a
 	github.com/mattn/go-isatty v0.0.8
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/mitchellh/go-wordwrap v1.0.0 // indirect
 	github.com/mitchellh/mapstructure v1.1.2
-	github.com/mxk/go-flowrate v0.0.0-20140419014527-cca7078d478f // indirect
-	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
-	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20190128024246-5eb7ae5bdb7a
+
+	// TODO: Can OLM use valid go modules version numbers?
+	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20190926160646-a61144936680
```

Comment:
We've been wanting to, but some openshift automation added a [`v3.11` tag](https://github.com/operator-framework/operator-lifecycle-manager/tree/v3.11.0) to our repository, which would force us to start at a greater semver for minimal version selection to skip it. IIRC, this happened right as we started to use modules (go 1.11 timeframe). We should revisit this soon.

Reply:
Oh no! That's a bummer. I don't envy you. However, since there's no `go.mod` file on the commit tagged with `v3.11.0`, you might not be totally hosed. See https://github.com/golang/go/wiki/Modules#semantic-import-versioning, specifically item 2 in the "Three Transitional Exceptions" section.

It might be worth trying to add a valid go module semver tag in a throwaway fork at `v0.12.0` to see whether you can import at that version without getting bumped "up" to `v3.11.0`. I'm pretty sure that will work.

Also since `v3.11.0` is not a go module (since it is lacking a `go.mod` file), you may even be able to delete the `v3.11.0` tag with little consequences. Don't quote me on that, but that's something else that might be worth trying in a throwaway fork.